source/client-side-encryption/tests/basic.template.yml

```diff
@@ -0,0 +1,42 @@
+database_name: &database_name "default"
+collection_name: &collection_name "default"
+
+encryption_setup:
+  collections:
+    - database: default
+      collection: default
+      data: []
+      jsonSchema: {{get_schema(0)}}
+    - database: admin
+      collection: datakeys
+      data: [{{get_key(0)}}]
+tests:
+  - description: "Insert a document requiring encryption" 
+    clientOptions:
+      clientSideEncryptionOptions:
+        kmsProvider: "aws-kms"
+        useRemoteSchemas: True
+    operations:
+      - name: insertOne
+        arguments:
+          document: &doc0 { _id: 1, ssn: "457-55-5462" }
+      - name: findOne
+        arguments:
+          filter: { _id: 1 }
+        result: *doc0
+    expectations:
+      - command_started_event:
```

Comment:
Because we're using the same MongoClient for listing collections and finding data keys, the events will have to reflect that, or else we'll have to develop a scheme for filtering them out.

Here's a gist with the actual events I'm getting when executing this test: https://gist.github.com/jyemin/eb14c657bac6f5934ec8a43344026957

I believe the extra events are due to `libmongocrypt` currently not doing any caching of schemas or data keys.  If so, these are the events I would actually expect: https://gist.github.com/jyemin/6b70fbdc375e768520726e81790349f7

One possible surprise here: because we're using the same MongoClient to look up the data keys, the driver naively tries to encrypt the find command sent to the key vault collection.  We may want to figure out a way to prevent that, though in practice it's probably not a big deal.